lib/active_merchant/billing/gateways/trust_commerce.rb

```diff
@@ -312,9 +312,9 @@ def scrub(transcript)
       private
 
       def add_aggregator(params, options)
-        if @options[:aggregator_id]
+        if @options[:aggregator_id] || application_id != Gateway.application_id
```

Comment:
I'm not following why this is `!=`...? Seems like it should just be `=`. This also means the test might not be sufficient. Unless I'm misunderstanding this...

Reply:
There is a default value for all Gateways. I think the value is `ActiveMerchant` or something like that.

Reply:
Well thats what this made me believe: https://github.com/activemerchant/active_merchant/blob/60efe5c5bd27cafa22edb6cdef2273ca7a8da72f/lib/active_merchant/billing/gateways/eway_rapid.rb#L45

But I can't find where its set.

Reply:
It's set to nil by default https://github.com/activemerchant/active_merchant/blob/6ea353a450fd48c1df7c539e698327bd5ac40e0d/lib/active_merchant/billing/gateway.rb#L142

Reply:
okay so yea there is a default value. We want to only use it if there's a different value.

Reply:
I sorted out my confusion, but in any case I don't think we need to check `Gateway`'s value.

Reply:
If it's not different, `application` would already have inherited the default value, right?

Reply:
Right so in that case we **don't** want to set `aggregatorID`

Reply:
Gah, got it. I don't know why this was so confusing for me but I'm better now. 👍